crates/worker/src/partition/leadership/mod.rs

```diff
@@ -317,7 +316,11 @@ where
 
             abort_result.map_err(Error::Invoker)?;
 
-            Self::unwrap_task_result(shuffle_result).map_err(Error::FailedShuffleTask)?;
+            if let Some(Err(err)) = shuffle_result {
+                if err.is_panic() {
+                    panic::resume_unwind(err.into_panic());
+                }
+            }
```

Comment:
Sounds like something worth leaving a comment about.

Reply:
Good point. Will add it.

Reply:
I think this can be simplified to an `expect` since the `TaskCenter` should already handle panicks originating from the `Shuffle` component.